src/joystick.cpp

```diff
@@ -343,20 +342,19 @@ bool joystick_manager::update_highlighted_hex(map_location& highlighted_hex) {
 	return true;
 }
 
-const map_location joystick_manager::get_direction(const map_location& loc, joystick_manager::DIRECTION direction) {
-
-	int x = loc.x;
-	int y = loc.y;
+const map_location joystick_manager::get_direction(const map_location& loc, joystick_manager::DIRECTION direction)
+{
+	map_location l = loc;
 
 	switch(direction) {
-		case NORTH:      return map_location(x, y - 1);
-		case SOUTH:      return map_location(x, y + 1);
-		case SOUTH_EAST: return map_location(x + 1, y + (1+is_odd(x))/2 );
-		case SOUTH_WEST: return map_location(x - 1, y + (1+is_odd(x))/2 );
-		case NORTH_EAST: return map_location(x + 1, y - (1+is_even(x))/2 );
-		case NORTH_WEST: return map_location(x - 1, y - (1+is_even(x))/2 );
-		case WEST:       return map_location(x - 1, y);
-		case EAST:       return map_location(x + 1, y);
+		case NORTH:      return l.get_direction(map_location::NORTH);
```

Comment:
Ah, I guess a switch is safer then. You could cast one to the other, but... best not to.